api/src/main/java/org/apache/iceberg/Scan.java

```diff
@@ -171,4 +172,10 @@ default ThisT select(String... columns) {
 
   /** Returns the split open file cost for this scan. */
   long splitOpenFileCost();
+
+  /** Create a new scan that will report the scan metrics to the {@code reporter} */
```

Comment:
The doc should indicate that this adds a reporter to the list of existing reporters, not overrides it.